scripts/generate-sponsorship-csv.ts

```diff
@@ -109,9 +139,13 @@ function getCurrentMonthFile(): string {
 }
 
 function main() {
-  const lastFourFiles = getLastNWeeklyFiles(4)
-  const weeklyData = lastFourFiles.map((file) => readWeeklyData(file))
-  const sponsorships = calculateSponsorship(weeklyData)
+  const lastFourWeeks = getLastNWeeklyFiles(4)
+  const weeklyData = lastFourWeeks.map(({ filePath, endDate }) => ({
+    endDate,
+    data: readWeeklyData(filePath),
+  }))
+  const weeklyDataAsc = weeklyData.reverse()
+  const sponsorships = calculateSponsorship(weeklyDataAsc)
```

Comment:
you MUST be idempotent (that is the whole point of the PR)

Consider the month you're generating for, then compute the dates of the weeks that are relevant for that month, then find the relevant contributionByUsers